### ISSUE TITLE
Feature Request: Treat nvim-treesitter-textobjects as an optional dependency

### ISSUE THREAD

**Issue by andrewferrier** [mawkler/demicolon.nvim#6]
demicolon looks really interesting, but it would be great if it could be decoupled from `nvim-treesitter-textobjects` so that was an (optional) extra dependency; I'm not interested in using `-textobjects`. It looks like it's a hard dependency which has to be installed right now.

**Comment by mawkler** [comment]
The core functionality of Demicolon is provided by nvim-treesitter-textobjects. What is the problem that you see with using nvim-treesitter-textobjects as a dependency?

**Comment by ColinKennedy** [comment]
I use this plugin primarily for Neovim's built-in pair-wise mappings. (e.g. `[q` / `]q`, `[b`, / `]b`, etc) so `nvim-treesitter-textobjects` isn't a factor. I do have `nvim-treesitter-textobjects` too anyway, FWIW.

**Comment by andrewferrier** [comment]
Yep, what @ColinKennedy said :) It's not a huge issue, don't get me wrong, but it seems like it's baked in as a hard dependency and I'm not sure if it needs to be. I won't use things like `[f`. `gitsigns` is optional, so it seems like `nvim-treesitter-textobjects` could be too...

**Comment by mawkler** [comment]
The logic for repeating any motion (i.e. both treesitter and non-treesitter motions) in Demicolon uses [nvim-treesitter-textobjects' motion repetition logic](https://github.com/nvim-treesitter/nvim-treesitter-textobjects/blob/master/lua/nvim-treesitter/textobjects/repeatable_move.lua). This means that decoupling Demicolon from nvim-treesitter-textobjects is non-trivial, and I don't really understand what the win would be.

**Comment by andrewferrier** [comment]
Ah, OK. That's an implementation detail, obviously, but I can understand the reluctance then. I wasn't aware of that. FWIW I would say not worth implementing given that restriction, so if you want to close, I'd understand.

The only win is 'cleanliness' and overall reducing the number of external dependencies.

Perhaps as a compromise it might be worth explaining this in the README?

**Comment by mawkler** [comment]
I've now clarified the dependency in the README. I'll close this issue now, but I'm open to accepting a PR that removes the nvim-treesitter-textobjects dependency. Thank you for your input!

**Comment by andrewferrier** [comment]
No problem, thanks for the plugin!
